boards/px4/fmu-v5/src/spi.cpp

```diff
@@ -0,0 +1,306 @@
+{
+	if (devid == SPIDEV_FLASH(0)) {
+		stm32_gpiowrite(GPIO_SPI2_CS_FRAM, !selected);
+	}
+}
+
+__EXPORT uint8_t stm32_spi2status(FAR struct spi_dev_s *dev, uint32_t devid)
+{
+	return SPI_STATUS_PRESENT;
+}
+#endif // CONFIG_STM32F7_SPI2 && GPIO_SPI2_CS_FRAM
+
+/************************************************************************************
+ * Name: stm32_spi4select and stm32_spi4status
+ *
+ * Description:
+ *   Called by stm32 spi driver on bus 4.
+ *
+ ************************************************************************************/
+#ifdef CONFIG_STM32F7_SPI4
+__EXPORT void stm32_spi4select(FAR struct spi_dev_s *dev, uint32_t devid, bool selected)
+{
+	ASSERT(PX4_SPI_BUS_ID(devid) == PX4_SPI_BUS_BARO);
+
+	// Making sure the other peripherals are not selected
+	for (auto cs : spi4selects_gpio) {
+		stm32_gpiowrite(cs, 1);
+	}
+
+	// SPI select is active low, so write !selected to select the device
+	stm32_gpiowrite(spi4selects_gpio[PX4_SPI_DEV_ID(devid)], !selected);
+}
+
+__EXPORT uint8_t stm32_spi4status(FAR struct spi_dev_s *dev, uint32_t devid)
+{
+	return SPI_STATUS_PRESENT;
+}
+#endif // CONFIG_STM32F7_SPI4
+
+/************************************************************************************
+ * Name: stm32_spi5select and stm32_spi5status
+ *
+ * Description:
+ *   Called by stm32 spi driver on bus 5.
+ *
+ ************************************************************************************/
+#ifdef CONFIG_STM32F7_SPI5
+__EXPORT void stm32_spi5select(FAR struct spi_dev_s *dev, uint32_t devid, bool selected)
+{
+	ASSERT(PX4_SPI_BUS_ID(devid) == PX4_SPI_BUS_EXTERNAL1);
+
+	// Making sure the other peripherals are not selected
+	for (auto cs : spi5selects_gpio) {
+		stm32_gpiowrite(cs, 1);
+	}
+
+	// SPI select is active low, so write !selected to select the device
+	stm32_gpiowrite(spi5selects_gpio[PX4_SPI_DEV_ID(devid)], !selected);
+}
+
+__EXPORT uint8_t stm32_spi5status(FAR struct spi_dev_s *dev, uint32_t devid)
+{
+	return SPI_STATUS_PRESENT;
+}
+#endif // CONFIG_STM32F7_SPI5
+
+/************************************************************************************
+ * Name: stm32_spi6select and stm32_spi6status
+ *
+ * Description:
+ *   Called by stm32 spi driver on bus 6.
+ *
+ ************************************************************************************/
+#ifdef CONFIG_STM32F7_SPI6
+__EXPORT void stm32_spi6select(FAR struct spi_dev_s *dev, uint32_t devid, bool selected)
+{
+	ASSERT(PX4_SPI_BUS_ID(devid) == PX4_SPI_BUS_EXTERNAL2);
+
+	// Making sure the other peripherals are not selected
+	for (auto cs : spi6selects_gpio) {
+		stm32_gpiowrite(cs, 1);
+	}
+
+	// SPI select is active low, so write !selected to select the device
+	stm32_gpiowrite(spi6selects_gpio[PX4_SPI_DEV_ID(devid)], !selected);
+}
+
+__EXPORT uint8_t stm32_spi6status(FAR struct spi_dev_s *dev, uint32_t devid)
+{
+	return SPI_STATUS_PRESENT;
+}
+#endif // CONFIG_STM32F7_SPI6
+
```

Comment:
Why not normalize the CONFIG_STM32F7_SPI6 and CONFIG_STM32F_SPI6 and 
with PX4_CONFIG_SPI1..N and then move this to board common using the hal gpio assessors. Then there would be one SPI CS driven only from BSP.

Reply:
Once this settles I'll think about moving it to a general (board common) location with micro hal usage.